### PR TITLE
validation: implement PosixHooksCalledInOrder test

### PR DESCRIPTION
**Pull request by q384566678**
Signed-off-by: Zhou Hao <zhouhao@cn.fujitsu.com>

### CODE REVIEW

**Comment by liangchenye** [comment]
1. I prefer to rename it to hooksorder.go .
2.  We need to verify 'poststart/poststop' as well. 
```    
   echo prestart1 > output_prestart
   echo prestart2 > output_prestart
   verify prestart hook order
   echo poststart1 > output_poststart
   echo poststart2 > output_poststart
   verify poststart hook order
   echo poststop1 > output_poststop
   echo poststop2 > output_poststop
   verify poststop hook order
```

**Comment by liangchenye** [comment]
I think in the order tests, we have the assumption: every hook ('echo') will be executed collect， we just test the 'order'.

**Comment by q384566678** [comment]
updated, and I think hook.go would be better.

File: validation/hooks.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+			r.SetID(uuid.NewV4().String())
+			g := util.GetDefaultGenerator()
+			output = filepath.Join(r.BundleDir, g.Spec().Root.Path, "output")
+			prestart1 := rspec.Hook{
+				Path: filepath.Join(r.BundleDir, g.Spec().Root.Path, "/bin/sh"),
+				Args: []string{
+					"sh", "-c", fmt.Sprintf("echo 'pre-start1 called' >> %s", output),
+				},
+			}
+			g.AddPreStartHook(prestart1)
```
Comment (wking) [inline validation/hooks.go]: No need for a `prestart1` temporary variable, you can just use:

```go
err := g.AddPreStartHook(rspec.Hook{
  …
})
if err != nil {
  return err
}
```

That also checks this call's returned error, which you're currently ignoring.

File: validation/hooks.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+	var output string
+	config := util.LifecycleConfig{
+		Actions: util.LifecycleActionCreate | util.LifecycleActionStart | util.LifecycleActionDelete,
+		PreCreate: func(r *util.Runtime) error {
+			r.SetID(uuid.NewV4().String())
+			g := util.GetDefaultGenerator()
+			output = filepath.Join(r.BundleDir, g.Spec().Root.Path, "output")
+			prestart1 := rspec.Hook{
+				Path: filepath.Join(r.BundleDir, g.Spec().Root.Path, "/bin/sh"),
```
Comment (wking) [inline validation/hooks.go]: You use this `Join` a lot in this test.  Maybe store it in a temporary variable:

```go
shPath := filepath.Join(r.BundleDir, g.Spec().Root.Path, "/bin/sh")
…
  Path: shPath,
```

Comment (q384566678) [reply validation/hooks.go]: updated, thanks.

**Comment by liangchenye** [comment]
LGTM

**Pull request merged**